optuna/study/_optimize.py

```diff
@@ -236,9 +245,14 @@ def _run_trial(
 
 
 def _log_failed_trial(
-    trial: FrozenTrial, message: Union[str, Warning], exc_info: Any = None
+    trial: FrozenTrial,
+    message: Union[str, Warning],
+    exc_info: Any = None,
+    value_or_values: Any = None,
 ) -> None:
     _logger.warning(
         "Trial {} failed because of the following error: {}".format(trial.number, message),
         exc_info=exc_info,
     )
+
+    _logger.warning("Trial {} failed with value {}.".format(trial.number, value_or_values))
```

Comment:
I think the message should distinct float and string, e.g., 1.23 and '1.23'.

```suggestion
    _logger.warning("Trial {} failed with value {}.".format(trial.number, repr(value_or_values)))
```

Reply:
Thank you for your suggestion!